server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java

```diff
@@ -26,7 +26,7 @@
 
     Mono<BlobId> save(byte[] data);
 
-    Mono<BlobId> save(InputStream data);
+    Mono<BlobId> save(InputStream data, long contentLength);
```

Comment:
Question, that is important in practice:  `does this contentLength needs to be exact?`

Can I safely pass to it a "too big value"?

Reply:
What would be the strategy when `zipping`? IE vault + backup?

Should we save stuff to a tmp file?

Reply:
1/ The content length processing depends on the ObjectStorage implementation.
From what I've seen:
- you can do what you want with Swift
- Scality is strict

2/ No idea, it has to be groomed.
May be the zipping process can give this metadata.

Reply:
> May be the zipping process can give this metadata.

Well you don't know the space it takes before zipping stuff... Temp files will then be compulsory...